### DISCUSSION TITLE
ejabberd Docs now using MkDocs

### DISCUSSION THREAD

**Discussion by badlop** [processone/ejabberd#4209] [Announcements]
> This article was originally published in: [ejabberd Docs now using MkDocs](https://www.process-one.net/blog/ejabberd-docs-now-using-mkdocs/)

The [ejabberd Docs](https://docs.ejabberd.im/) website did just get a major rework: new content management system, reorganized navigation, improved markdown, and several improvements!

## Brief documentation timeline

ejabberd started in November 2002 (see a timeline in the [ejabberd turns 20](https://www.process-one.net/blog/ejabberd-turns-20/) blog post). And the first documentation was published in January 2003, using LaTeX, see [Ejabberd Installation and Operation Guide](https://web.archive.org/web/20030409163941/http://ejabberd.jabberstudio.org/guide.html). That was one single file, hosted in the ejabberd CVS source code repository, and was available as a single HTML file and a PDF.

As the project grew and got more content, in 2015 the documentation was [converted from LaTeX to Markdown](https://www.process-one.net/blog/ejabberd-new-documentation-site-a-community-effort/), moved from ejabberd repository to a dedicated [docs.ejabberd.im](https://github.com/processone/docs.ejabberd.im) git repository, and published using a Go HTTP server in `docs.ejabberd.im`, see an [archived ejabberd Docs site](https://web.archive.org/web/20240327173114/https://docs.ejabberd.im/).

## New ejabberd Docs site

Now the ejabberd documentation has moved to MkDocs+Material, and this brings several changes and improvements:

#### Site and Web Server:

- Replaced Go site with [MkDocs](https://www.mkdocs.org/)
- [Material](https://squidfunk.github.io/mkdocs-material/) theme for great features and visual appeal, including light/dark color schemes
- Still written in Markdown, but now using several MkDocs, Material and [Python-Markdown](https://python-markdown.github.io/extensions/) extensions
- The online site is built by GitHub Actions and hosted in Pages, with smaller
automatic deployment time
- Offline reading: the `ejabberd Docs` site can be downloaded as a PDF or zipped HTML, see the links in [home page](https://docs.ejabberd.im/)

#### Navigation

- Major navigation reorganization, keeping URLs intact so old links still work (only Install got some relevant URL changes)
- [Install](https://docs.ejabberd.im/admin/install/) section is split into several sections: Containers, Binaries, Compile, ...
- Reorganized the [Archive](https://docs.ejabberd.im/archive/) section, and now it includes the corresponding Upgrade notes
- Several markdown files from the ejabberd and docker-ejabberd repositories are now incorporated here

#### Content

- Many markdown visual improvements, specially in code snippets
- Options and commands that were modified in the last release will show a mark, see for example [API Reference](https://docs.ejabberd.im/developer/ejabberd-api/admin-api/)
- Version annotations are shown after the corresponding title, see for example [sql_flags](https://docs.ejabberd.im/admin/configuration/toplevel/#sql_flags)
- Modules can have version annotations, see for example [mod_matrix_gw](https://docs.ejabberd.im/admin/configuration/modules/#mod_matrix_gw)
- Links to modules, options and API now use the real name with `_` character instead of `-` (compare old [#auth-opts](https://web.archive.org/web/20231220132058/https://docs.ejabberd.im/admin/configuration/toplevel/#auth-opts) with [#auth_opts](https://docs.ejabberd.im/admin/configuration/toplevel/#auth_opts)). The old links are still supported, no broken links.
- [Listen Modules](https://docs.ejabberd.im/admin/configuration/listen/) section is now better organized
- New experimental [ejabberd Developer Livebook](https://docs.ejabberd.im/livebooks/ejabberd-developer-livebook/)

So, please check the revamped [ejabberd Docs](https://docs.ejabberd.im/) site, and head to [docs.ejabberd.im git repository](https://github.com/processone/docs.ejabberd.im) to report problems and propose improvements.

**Comment by dalakatt** [comment]
See the future of AI with OpenText World Europe keynotes.
Watch on-demand now


Solutions
Smarter document management for life sciences
Build a better-connected drug development value chain with smarter document management
On this page:
Achieve operational excellence through smarter document management

Ensure compliance during manufacturing processes and improve submission quality
Bringing safe, high-quality products to market faster—and at a lower cost—requires unifying processes that extend across domains, divisions, and external partners. Proper document management is critical for life sciences companies to avoid regulatory non-compliance fines, delayed product launches, and product recalls. Robust document lifecycle management can improve efficiency, boost compliance, and contribute to faster drug and device development.

Ready to fast-track your document processes with smarter document management?
The benefits of document management for life sciences
Strengthen document control and boost operational efficiency.


Improve audit readiness to minimize compliance risk
Demonstrate FDA 21 CFR Part 11 and EMA Annex 11 compliance and ensure documents are readily available for audits, lowering the risk of regulatory non-compliance.


Increase control and flexibility for document-related tasks
Leverage document taxonomies, lifecycles, and workflows for pharmaceuticals and medical devices to enforce corporate document controls while enhancing local productivity.


Unify document control to ensure compliance and speed time to market
Centralize document management to enable efficient coordination between departments, improve collaboration, break down information silos, and foster transparency.


Foster compliant, secure collaboration
Collaborate and reuse content and processes with internal and remote employees, authorized partners, and other third parties without compromising security or compliance.


Discover and understand content faster
Use an AI-powered digital assistant to empower workers with conversational search, content discovery, and summarization to accelerate processes across the value chain.


Archive intelligently and compliantly
Move aging data to a cloud archive service during SAP S/4HANA migration to reduce costs. Decommission obsolete applications to remove silos and modernize infrastructure.

How document management and control can impact your business
Higher product quality in manufacturing processes
Manufacturers must protect their profits and reputation. Production stoppages for non-compliance can negatively impact both. Managing documents with Good Manufacturing Practices (GMP) standards ensures consistent processes and higher product quality.

Submission readiness with controlled content
Health agency submissions require a high degree of accuracy. Delayed regulatory filings due to rework can cost millions of dollars per delay. Ensure proper authoring, collaboration, and approval workflows with a document lifecycle management system.

Inspection-ready trial master files (TMFs)
Timely completion of clinical trial master files is essential. Delays can cost millions daily in lost revenue. File planning templates and workflows facilitate efficient completion of eTMFs while dashboards provide real-time insights.

Audit-readiness and compliance risk mitigation
Incomplete or inaccurate information increases compliance risk. Ensure audit readiness and reduce compliance risk by managing documentation within pre-validated GxP environments aligned with FDA 21 CFR Part 11 and EMA standards.

A single source of regulated content
Life sciences workers need continuous access to the latest content. Duplicate documents and outdated versions can cause confusion. Manage content and processes in a unified source, based on roles and permissions, to enhance efficiency and compliance.

Effective enterprise document controls
The pharmaceuticals and medical devices sector relies on document integrity. Inadequate controls can lead to data manipulation. Leverage document taxonomies, lifecycles, and workflows specifically designed for the sector to minimize these risks.
